### PR TITLE
Add Fastly clearing

### PR DESCRIPTION
**Pull request by thomasleese**
This adds the ability to clear Fastly cache when receiving a message on the queue.

Ready for review, but I've marked as do not merge because I think we should wait until we've fully tested the Varnish caching in case anything weird comes up before we start clearing Fastly.

[Trello Card](https://trello.com/c/9k1peA1F/434-make-the-new-cache-clearing-app-invalidate-cache-in-fastly)

### CODE REVIEW

File: spec/purger_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  let(:url) { "https://www.gov.uk/government/news/govuk-implements-new-cache-clearing" }
+
+  subject { described_class.new(Logger.new("/dev/null")) }
+
+  context "if the purge request succeeds" do
+    before do
+      stub_request(:purge, /cache/)
+    end
+
+    it "purges the varnish cache for the base path in the payload" do
```
Comment (elliotcm) [inline spec/purger_spec.rb]: Should this now be generic, i.e. not "varnish"?

**Comment by thomasleese** [comment]
@elliotcm I like the idea of the tests checking that the final request is made and not caring what does it, but in that sense they're more integration tests. I see what you mean though, I'll have a good at refactoring them a bit.

File: spec/purger_spec.rb (hunk 1 above)
Comment (thomasleese) [reply spec/purger_spec.rb]: Yep, I'll fix that.

**Comment by thomasleese** [comment]
@elliotcm I've refactored the tests to be more like unit tests.

File: app/processor.rb [hunk 2]
```diff
@@ -4,15 +4,20 @@
...
+    @varnish_clearer = VarnishClearer.new(logger)
+    @fastly_clearer = FastlyClearer.new(logger)
   end
 
   def process(message)
-    VarnishClearer.new(logger).clear_for(message.payload.fetch("base_path"))
+    base_path = message.payload.fetch("base_path")
+
+    varnish_clearer.clear_for(base_path)
+    fastly_clearer.clear_for(base_path)
```
Comment (kevindew) [inline app/processor.rb]: Should you be clearing all the items in the routes array rather than base_path?

Comment (thomasleese) [reply app/processor.rb]: Good point, I think we'll add that in a separate PR.

**Pull request merged**